src/spi.rs

```diff
@@ -414,14 +438,60 @@ impl<'d> SpiMasterDriver<'d> {
             quadwp_io_num: -1,
             quadhd_io_num: -1,
 
-            max_transfer_sz: config.dma.max_transfer_size() as i32,
+            max_transfer_sz: max_transfer_sz as i32,
             ..Default::default()
         };
 
-        esp!(unsafe { spi_bus_initialize(SPI::device(), &bus_config, config.dma.into()) })?;
+        esp!(unsafe { spi_bus_initialize(SPI::device(), &bus_config, dma_chan) })?;
+        Ok(max_transfer_sz)
+    }
+}
+
+impl<'d> Drop for SpiMasterDriver<'d> {
+    fn drop(&mut self) {
+        // should we drop without checking the lock?
+        let guard = self.lock.enter();
+        esp!(unsafe { spi_bus_free(self.handle) }).unwrap();
+    }
+}
+
+pub struct EspSpiDevice<'d, T> {
+    handle: RefCell<Option<spi_device_handle_t>>,
+    cs_pin_ref: PeripheralRef<'d, AnyOutputPin>,
+    config: config::Config,
+    con: spi_device_interface_config_t,
```

Comment:
Do these really need to be stored? The user can store them outside if necessary, no?

Reply:
i commented out the convenience fn's and rm the config store. still storing spi_device_interface_config_t to allow the user  to manual use  rm_from_bus or add_to_bus without needing to reconstruct an EspSpiDevice to just update the device config. but yeah i can cut that out also if not needed. we would need an update_conf fn that manipulates the stored spi_device_interface_config_t though to allow it to work